sdrbase/util/aprs.cpp

```diff
@@ -988,3 +1013,196 @@ bool APRSPacket::parseTelemetry(QString& info, int& idx)
     else
         return false;
 }
+
+// Mic-E Implementation by Peter Beckman KM4BBB github:ooglek
+bool APRSPacket::parseMicE(QString& info, int& idx, QString& dest)
+{
+    info = info.toLatin1();
+    // Mic-E Location data is encoded in the AX.25 Destination Address
+    if (dest.length() < 6) {
+        qDebug() << "APRSPacket::parseMicE: Destination invalid length " << dest;
+        return false;
+    }
+
+    // Mic-E Information data is 8 bytes minimum, 13-14 with altitude
+    if (info.length() < idx+8) {
+        qDebug() << "APRSPacket::parseMicE: Information Data invalid length " << info;
+        return false;
+    }
+
+    QString latDigits = "";
+    QString messageBits = "";
+    int messageType = 0; // 0 = Standard, 1 = Custom
+    int longitudeOffset = 0;
+    float latitudeDirection = 1; // Assume North
+    float longitudeDirection = -1; // Assume West
+
+    QHash<QString, QString> messageTypeLookup;
+    messageTypeLookup["111"] = "Off Duty";
+    messageTypeLookup["110"] = "En Route";
+    messageTypeLookup["101"] = "In Service";
+    messageTypeLookup["100"] = "Returning";
+    messageTypeLookup["011"] = "Committed";
+    messageTypeLookup["010"] = "Special";
+    messageTypeLookup["001"] = "Priority";
+    messageTypeLookup["000"] = "Emergency";
```

Comment:
Is there a more compact way to define a hash???

Reply:
To initialize a QHash with a literal definition you would do something like this:
```c++
	QHash<QString, QString> messageTypeLookup = {
		{"011", "Test"},
		{"001", "Hello"}
	};
```
Not necessarily more compact but somehow more elegant